app/src/main/java/ch/hikemate/app/ui/components/BackButton.kt

```diff
@@ -27,9 +27,13 @@ object BackButton {
  * @param navigationActions The navigation actions.
  */
 @Composable
-fun BackButton(navigationActions: NavigationActions, modifier: Modifier = Modifier) {
+fun BackButton(
+    navigationActions: NavigationActions,
+    modifier: Modifier = Modifier,
+    onClick: () -> Unit = { navigationActions.goBack() }
+) {
   IconButton(
-      onClick = { navigationActions.goBack() },
+      onClick = { onClick() },
```

Comment:
Question: This is more of a general fix, or? Or does it have something to do with the asynchronous operations?

Reply:
It's just a fix to allow actions to be taken when clicking on the goBack button, other than going back (for example, clearing the hike selected)